File: CS/ProgressTrackBar/MySkinTrackBarObjectPainter.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraEditors.Drawing;
using DevExpress.Skins;
using System.Drawing;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    class MySkinTrackBarObjectPainter : SkinTrackBarObjectPainter
    {
        public MySkinTrackBarObjectPainter(ISkinProvider provider) : base(provider) { }
        protected override void DrawLine(TrackBarObjectInfoArgs e, System.Drawing.Point p1, System.Drawing.Point p2)
        {
            base.DrawLine(e, p1, p2);
        }
        public override void DrawTrackLine(TrackBarObjectInfoArgs e)
        {
            Rectangle rec = e.ViewInfo.TrackBarHelper.Rotate(e.ViewInfo.TrackLineRect);
            rec.X = e.ViewInfo.PointsRect.X;
            rec.Width = e.ViewInfo.PointsRect.Width;

            ProgressBarControl prb = (e.ViewInfo.Item as MyRepositoryItemTrackBar).Progress;
            if (prb == null)
            {
                base.DrawTrackLine(e);
                return;
            }
            BaseControlViewInfo vi = prb.GetViewInfo();
            vi.CalcViewInfo(e.Graphics, MouseButtons.None, Point.Empty, rec);
            ControlGraphicsInfoArgs args = new ControlGraphicsInfoArgs(vi, e.Cache, rec);
            vi.Painter.Draw(args);

        }
    }
}

File: VB/ProgressTrackBar/MySkinTrackBarObjectPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.Skins
Imports System.Drawing
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports System.Windows.Forms

Namespace WindowsFormsApplication1
	Friend Class MySkinTrackBarObjectPainter
		Inherits SkinTrackBarObjectPainter
		Public Sub New(ByVal provider As ISkinProvider)
			MyBase.New(provider)
		End Sub
		Protected Overrides Sub DrawLine(ByVal e As TrackBarObjectInfoArgs, ByVal p1 As System.Drawing.Point, ByVal p2 As System.Drawing.Point)
			MyBase.DrawLine(e, p1, p2)
		End Sub
		Public Overrides Sub DrawTrackLine(ByVal e As TrackBarObjectInfoArgs)
			Dim rec As Rectangle = e.ViewInfo.TrackBarHelper.Rotate(e.ViewInfo.TrackLineRect)
			rec.X = e.ViewInfo.PointsRect.X
			rec.Width = e.ViewInfo.PointsRect.Width

			Dim prb As ProgressBarControl = (TryCast(e.ViewInfo.Item, MyRepositoryItemTrackBar)).Progress
			If prb Is Nothing Then
				MyBase.DrawTrackLine(e)
				Return
			End If
			Dim vi As BaseControlViewInfo = prb.GetViewInfo()
			vi.CalcViewInfo(e.Graphics, MouseButtons.None, Point.Empty, rec)
			Dim args As New ControlGraphicsInfoArgs(vi, e.Cache, rec)
			vi.Painter.Draw(args)

		End Sub
	End Class
End Namespace
